ark/analysis/spatial_analysis.py

```diff
@@ -169,71 +173,73 @@ def calculate_channel_spatial_enrichment(dist_matrices_dict, marker_thresholds,
         context_names = all_data[context_col].unique()
         context_pairings = combinations_with_replacement(context_names, 2)
 
-    for fov in included_fovs:
-        # Subsetting expression matrix to only include patients with correct fov label
-        current_fov_idx = all_data[fov_col] == fov
-        current_fov_data = all_data[current_fov_idx]
+    # Subsetting expression matrix to only include patients with correct fov label
+    current_fov_idx = all_data[fov_col] == fov
+    current_fov_data = all_data[current_fov_idx]
 
-        # Patients with correct label, and only columns of channel markers
-        current_fov_channel_data = all_channel_data[current_fov_idx]
+    # Patients with correct label, and only columns of channel markers
+    current_fov_channel_data = all_channel_data[current_fov_idx]
 
-        # Retrieve fov-specific distance matrix from distance matrix dictionary
-        dist_matrix = dist_matrices_dict[fov]
+    # Get close_num and close_num_rand
+    close_num, channel_nums, mark_pos_labels = spatial_analysis_utils.compute_close_cell_num(
+        dist_mat=dist_matrix, dist_lim=dist_lim, analysis_type="channel",
+        current_fov_data=current_fov_data, current_fov_channel_data=current_fov_channel_data,
+        thresh_vec=thresh_vec, cell_label_col=cell_label_col
+    )
 
-        # Get close_num and close_num_rand
-        close_num, channel_nums, mark_pos_labels = spatial_analysis_utils.compute_close_cell_num(
-            dist_mat=dist_matrix, dist_lim=dist_lim, analysis_type="channel",
-            current_fov_data=current_fov_data, current_fov_channel_data=current_fov_channel_data,
-            thresh_vec=thresh_vec, cell_label_col=cell_label_col)
+    if context_col is not None:
+        close_num_rand = np.zeros((*close_num.shape, bootstrap_num), dtype=np.uint16)
 
-        if context_col is not None:
-            close_num_rand = np.zeros((*close_num.shape, bootstrap_num), dtype=np.uint16)
+        context_nums_per_id = \
+            current_fov_data.groupby(context_col)[cell_label_col].apply(list).to_dict()
 
-            context_nums_per_id = \
-                current_fov_data.groupby(context_col)[cell_label_col].apply(list).to_dict()
+        for name_i, name_j in context_pairings:
```

Comment:
I'm confused by this. This question probably doesn't belong in this PR, but why are we iterating through pairs of contexts? After Adam left, is there a go-to person for questions about this spatial enrichment script?

Reply:
Nope, not really